### PR TITLE
[16.0][ADD] account_analytic_category (replace account.analytic.group)

### PR DESCRIPTION
**Pull request by norlinhenrik**
2 new modules with 2 new fields on `account.move.line` to use in pivot:

- `account_move_line_analytic_account_ids`: new field `analytic_account_ids`.
- `account_analytic_category`: new field `analytic_category_ids`.

### CODE REVIEW

File: account_analytic_category/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+This module introduces account.analytic.category
+as a replacement for account.analytic.group which was removed in Odoo 16.0.
```
Comment (yostashiro) [inline account_analytic_category/readme/DESCRIPTION.rst:2]: Is there a reason to rename the model?

File: account_analytic_category/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+{
+    "name": "Analytic Categories",
+    "version": "16.0.1.0.0",
+    "author": "Henrik Norlin, Odoo Community Association (OCA)",
+    "category": "Account",
+    "website": "https://github.com/OCA/account-analytic",
+    "depends": [
+        "account_financial_report",  # account.move.line.analytic_account_ids
```
Comment (yostashiro) [inline account_analytic_category/__manifest__.py]: I thought this dependency was a bit heavy.  Maybe the part that depends on `account.move.line.analytic_account_ids` can be handled in a separate glue module in the account-financial-reporting repo.

Comment (norlinhenrik) [reply account_analytic_category/__manifest__.py]: I made a new module `account_move_line_analytic_account_ids`.

File: account_analytic_category/readme/DESCRIPTION.rst (hunk 1 above)
Comment (norlinhenrik) [reply account_analytic_category/readme/DESCRIPTION.rst:2]: Odoo called the model for [Analytic Categories](https://github.com/odoo/odoo/blob/15.0/addons/analytic/models/analytic_account.py#L37) and I thought maybe this is a better name. But I am flexible on this.

File: account_analytic_category/__manifest__.py (hunk 2 above)
Comment (yostashiro) [reply account_analytic_category/__manifest__.py]: There is also this PR https://github.com/OCA/account-analytic/pull/565 but the proposed solution in this PR doesn't store `analytic_account_ids` due to a technical reason.

Comment (norlinhenrik) [reply account_analytic_category/__manifest__.py]: Nice :) I have added `analytic_mixin_analytic_account` as a dependeny.

**Comment by norlinhenrik** [comment]
Waiting to merge `analytic_mixin_analytic_account`, see #565.

**Comment by norlinhenrik** [comment]
Maybe a better implementation would be to create an `analytic_category_distribution` JSON field on the `account.move.line`?

EDIT
Two observations in 16.0:
- `account.analytic.plan` has `parent_id`. Having one parent for all plans improves the UI.
- `account.move.line` has `analytic_account_id`. In pivot view, I can group by Analytic Account.

`account.move.line` with `analytic_plan_id` is really what I am missing. I would like to group by Analytic Plan.

**Pull request closed without merging**